libs/remix-lib/src/execution/txExecution.ts

```diff
@@ -75,20 +75,20 @@ export function checkVMError (execResult, compiledContracts) {
     error: false,
     message: ''
   }
-  if (!execResult.exceptionError) {
+  if (!execResult.exceptionError && !isInjectedWeb3) {
```

Comment:
seems like `error` and `returnData` are both the same but present under different properties... could you add a type `execResult`, so you don't have to do stuff like `isInjectedWeb3 ? execResult.error : exceptionError`

Reply:
that way you may not need to have the `isInjected`

Reply:
I think `execResult` is quite different for VM and injected Web3. For Injected web3, it is just an object having error details

Reply:
updated